src/text/span.rs

```diff
@@ -332,6 +332,37 @@ impl<'a> Span<'a> {
     pub fn to_right_aligned_line(self) -> Line<'a> {
         self.into_right_aligned_line()
     }
+
+    /// Splits the span into two at the given width.
+    ///
+    /// The first span will contain the characters that fit within the given width, and the second
+    /// span will contain the remaining characters.
+    pub(crate) fn split_at_width(&'a self, width: usize) -> (Self, Self) {
+        // There's at least two ways to do this. I chose the second option as it iterates over the
+        // string once at the cost of always allocating two strings regardless of the split point.
+        // 1. iterate the chars to find the split point and then split the spans.
+        // 2. iterate the chars and build the spans at the same time.
+        let mut left = String::new();
+        let mut right = String::new();
+        let mut left_width = 0;
+        for c in self.content.chars() {
```

Comment:
I _think_ this will cause issues with flag emojis

```rust
    let s = "🇺🇸1234";
    dbg!(s);
    dbg!(s.len());
    dbg!(s.chars().count());
    dbg!(s.width());
    dbg!(s.graphemes(true).count());
```

gives me

```
[src/main.rs:43:5] s = "🇺🇸1234"
[src/main.rs:44:5] s.len() = 12
[src/main.rs:45:5] s.chars().count() = 6
[src/main.rs:46:5] s.width() = 6
[src/main.rs:47:5] s.graphemes(true).count() = 5
```

I haven't thought about it extensively but can we do this instead? 

```suggestion 
        for s in self.content.graphemes(true) {
            for c in s.chars() {
```
      
```

Reply:
Maybe it is fine? It'll get split like this:

```
[src/main.rs:50:9] c = '🇺'
[src/main.rs:50:9] c = '🇸'
```

So it still won't panic.

Reply:
Oops, this is the same conversation that is happening on the main thread. Ignore these comments.

Reply:
Can you try `1234🇺🇸`?

Reply:
No difference
```rust
#[case::flag_1(1, "1")]
        #[case::flag_2(2, "12")]
        #[case::flag_3(3, "123")]
        #[case::flag_4(4, "1234")]
        #[case::flag_5(5, "1234 ")]
        #[case::flag_6(6, "1234🇺🇸")]
        #[case::flag_7(7, "1234🇺🇸 ")]
        #[case::flag_8(8, "1234🇺🇸  ")]
        #[case::flag_9(9, "1234🇺🇸   ")]
```